streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -515,6 +580,8 @@
      * @param windows       the specification of the aggregation {@link Windows}
      * @param keySerde      key serdes for materializing the counting table,
      *                      if not specified the default serdes defined in the configs will be used
+     *
```

Comment:
See my above comments, I think this is applicable for windowed agg / reduce also, just add "for each window" in the end.